src/mapbox/create-ref.ts

```diff
@@ -46,12 +48,20 @@ export default function createRef(mapInstance: Mapbox, mapLib: any): MapRef {
     getPadding: () => mapInstance.transform.padding,
     getBounds: () => mapInstance.transform.getBounds(),
     project: (lnglat: LngLatLike) => {
-      return mapInstance.transform.locationPoint(mapLib.LngLat.convert(lnglat));
+      const tr = map.transform;
+      map.transform = mapInstance.transform;
+      const result = map.project(lnglat);
+      map.transform = tr;
+      return result;
     },
     unproject: (point: PointLike) => {
-      return mapInstance.transform.pointLocation(mapLib.Point.convert(point));
+      const tr = map.transform;
+      map.transform = mapInstance.transform;
+      const result = map.unproject(point);
+      map.transform = tr;
+      return result;
     },
-    queryTerrainElevation: (lnglat: LngLatLike, options: ElevationQueryOptions) => {
+    queryTerrainElevation: (lnglat: LngLatLike, options: any) => {
```

Comment:
Is there no good type for this?

Reply:
This is where mapbox and maplibre diverge. I prefer not to type it in a way that we need to upgrade whenever they make a new release.